float32 errorX
float32 errorY
float32 errorT
float32 errorS


